doctr/models/recognition/crnn/tensorflow.py

```diff
@@ -74,22 +76,35 @@ def __call__(
         # Decode CTC
         _decoded, _log_prob = tf.nn.ctc_beam_search_decoder(
             tf.transpose(logits, perm=[1, 0, 2]),
-            tf.fill(logits.shape[0], logits.shape[1]),
-            beam_width=1, top_paths=1,
+            tf.fill(tf.shape(logits)[:1], tf.shape(logits)[1]),
+            beam_width=beam_width, top_paths=top_paths,
         )
-        out_idxs = tf.sparse.to_dense(_decoded[0], default_value=len(self.vocab))
-        probs = tf.math.exp(tf.squeeze(_log_prob, axis=1))
+
+        _decoded = tf.sparse.concat(1,
+                                    [tf.sparse.expand_dims(dec,axis=1) for dec in _decoded],
+                                    expand_nonconcat_dims=True) # dim : batchsize x beamwidth x actual_max_len_predictions
+        out_idxs = tf.sparse.to_dense(_decoded, default_value=len(self.vocab))
 
         # Map it to characters
         _decoded_strings_pred = tf.strings.reduce_join(
             inputs=tf.nn.embedding_lookup(tf.constant(self._embedding, dtype=tf.string), out_idxs),
             axis=-1
         )
         _decoded_strings_pred = tf.strings.split(_decoded_strings_pred, "<eos>")
-        decoded_strings_pred = tf.sparse.to_dense(_decoded_strings_pred.to_sparse(), default_value='not valid')[:, 0]
-        word_values = [word.decode() for word in decoded_strings_pred.numpy().tolist()]
+        decoded_strings_pred = tf.sparse.to_dense(_decoded_strings_pred.to_sparse(),
+                                                  default_value='not valid')[:,:,0] # dim : batch_size x beam_width
+
+        
+        if top_paths==1 :
+            probs = tf.math.exp(tf.squeeze(_log_prob, axis=1))# dim : batchsize
+            decoded_strings_pred = tf.squeeze(decoded_strings_pred,axis=1)
+        else :
+            probs = tf.math.exp(_log_prob)# dim : batchsize x beamwidth
 
-        return list(zip(word_values, probs.numpy().tolist()))
+        return {
```

Comment:
You are returning a dictionary containing tf.Tensor objects, I think we should keep the same output signature and return a list of string words, otherwise we are going to break the whole pipeline

Reply:
yes this signature change has been removed to keep only changes for allowing beam width